proxy/src/main/java/com/velocitypowered/proxy/connection/MinecraftConnection.java

```diff
@@ -279,27 +278,27 @@ public void setAssociation(MinecraftConnectionAssociation association) {
     this.association = association;
   }
 
-  public boolean isLegacyForge() {
-    return isLegacyForge;
-  }
-
-  public void setLegacyForge(boolean isForge) {
-    this.isLegacyForge = isForge;
-  }
-
-  public boolean canSendLegacyFmlResetPacket() {
-    return canSendLegacyFmlResetPacket;
-  }
-
-  public void setCanSendLegacyFmlResetPacket(boolean canSendLegacyFMLResetPacket) {
-    this.canSendLegacyFmlResetPacket = isLegacyForge && canSendLegacyFMLResetPacket;
-  }
-
   public int getNextProtocolVersion() {
     return this.nextProtocolVersion;
   }
 
   public void setNextProtocolVersion(int nextProtocolVersion) {
     this.nextProtocolVersion = nextProtocolVersion;
   }
+
+  /**
+   * Gets the detected {@link ConnectionType}
+   * @return The {@link ConnectionType}
+   */
+  public ConnectionType getConnectionType() {
+    return connectionType;
+  }
+
+  /**
+   * Sets the detected {@link ConnectionType}
+   * @param connectionType The {@link ConnectionType}
+   */
+  public void setConnectionType(ConnectionType connectionType) {
```

Comment:
You can drop `Connection` from the name - `connection.getType()` is clear enough, don't need `connection.getConnectionType()`.